### ISSUE TITLE
使用Native Backend测试Qwen2.5-14B-Instruct模型，MMLU得分很低

### ISSUE THREAD

**Issue by Dr-Corgi** [modelscope/evalscope#479]
## 问题描述 / Issue Description

使用Native Backend测试Qwen2.5-14B-Instruct模型，MMLU得分很低

## EvalScope 版本 / Version (必须)
```
>>> import evalscope
>>> evalscope.__version__
'2.0.0'
```

## 使用的工具 / Tools Used
- [√] Native / 原生框架
- [ ] Opencompass backend
- [ ] VLMEvalKit backend
- [ ] RAGEval backend
- [ ] Perf / 模型推理压测工具
- [ ] Arena  /竞技场模式

## 执行的代码或指令 / Code or Commands Executed

```python
from evalscope.run import run_task
from evalscope.config import TaskConfig

task_cfg = TaskConfig(
    model="Qwen/Qwen2.5-14B-Instruct",
    model_id="Qwen2.5-14B-Instruct",
    datasets=['mmlu'],
    generation_config={
        "temperature": 0.3,
        "top_k": 20,
        "top_p": 0.85,
        "do_sample": True
    }
)
run_task(task_cfg=task_cfg)
```

```bash
# 例如：在终端中执行的指令 / Terminal command executed
python3 run_native.py
```

## 运行环境 / Runtime Environment

- 操作系统 / Operating System:
  - [ ] Windows
  - [ ] macOS
  - [√] Ubuntu

- Python版本 / Python Version:
  - [√] 3.12.7
  - [ ] 3.11
  - [ ] 3.10
  - [ ] 3.9

## 其他信息 / Additional Information
测试MMLU结果如下：
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Model                | Dataset   | Metric          | Subset                              |   Num |   Score | Cat.0          |
+======================+===========+=================+=====================================+=======+=========+================+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_european_history        |   164 |  0.2561 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_us_history              |   203 |  0.2906 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_world_history           |   236 |  0.3136 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | philosophy                          |   310 |  0.2097 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | prehistory                          |   323 |  0.2508 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | logical_fallacies                   |   162 |  0.2593 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | formal_logic                        |   125 |  0.376  | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | international_law                   |   120 |  0.3583 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | professional_law                    |  1533 |  0.2505 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | jurisprudence                       |   107 |  0.2897 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | world_religions                     |   170 |  0.3765 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | moral_scenarios                     |   894 |  0.2383 | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | moral_disputes                      |   345 |  0.258  | Humanities     |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | business_ethics                     |    99 |  0.3838 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | clinical_knowledge                  |   264 |  0.2841 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | medical_genetics                    |    99 |  0.3333 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | virology                            |   165 |  0.2788 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | professional_accounting             |   281 |  0.242  | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | professional_medicine               |   271 |  0.2214 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | nutrition                           |   305 |  0.2656 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | global_facts                        |    99 |  0.2121 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | anatomy                             |   134 |  0.2164 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | college_medicine                    |   172 |  0.2209 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | human_aging                         |   222 |  0.3739 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | miscellaneous                       |   782 |  0.257  | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | marketing                           |   233 |  0.309  | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | management                          |   102 |  0.1961 | Other          |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_physics                 |   150 |  0.1933 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | college_computer_science            |    99 |  0.2929 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_biology                 |   309 |  0.2006 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | abstract_algebra                    |    99 |  0.2525 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | machine_learning                    |   111 |  0.3333 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | college_chemistry                   |    99 |  0.202  | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | elementary_mathematics              |   377 |  0.244  | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | college_mathematics                 |    99 |  0.2323 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_statistics              |   215 |  0.1721 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_mathematics             |   269 |  0.3011 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_computer_science        |    99 |  0.4848 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | conceptual_physics                  |   234 |  0.2906 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_chemistry               |   202 |  0.1931 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | college_physics                     |   101 |  0.2178 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | computer_security                   |    99 |  0.3333 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | electrical_engineering              |   144 |  0.2917 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | astronomy                           |   151 |  0.2252 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | college_biology                     |   143 |  0.2797 | STEM           |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_microeconomics          |   237 |  0.2152 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | econometrics                        |   113 |  0.3274 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | security_studies                    |   244 |  0.1885 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | public_relations                    |   109 |  0.211  | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | professional_psychology             |   611 |  0.3535 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | human_sexuality                     |   130 |  0.3538 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_government_and_politics |   192 |  0.2188 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_geography               |   197 |  0.203  | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_psychology              |   544 |  0.2022 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | sociology                           |   200 |  0.415  | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | us_foreign_policy                   |    99 |  0.3838 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+
| Qwen2.5-14B-Instruct | mmlu      | AverageAccuracy | high_school_macroeconomics          |   389 |  0.2339 | Social Science |
+----------------------+-----------+-----------------+-------------------------------------+-------+---------+----------------+

**Comment by Yunnglin** [comment]
本地加载模型在mmlu上评测默认使用的是logits进行答案判断（但prompt不适配），现已改为generation的方式，可以拉main分支代码试试

**Comment by Dr-Corgi** [comment]
> 本地加载模型在mmlu上评测默认使用的是logits进行答案判断（但prompt不适配），现已改为generation的方式，可以拉main分支代码试试

拉取了最新代码，经测试符合预期了
